pages/templates/pages/microsite_page.html

```diff
@@ -1,12 +1,210 @@
 {% extends "pages/base_page.html" %}
 
-{% load wagtailcore_tags %}
+{% load wagtailcore_tags wagtailimages_tags %}
+
+{% block meta %}
+  {{ block.super }}
+  <style>
+    #custom_header {
+      {% if page.custom_header_background_color %}
+        background-color: #{{ page.custom_header_background_color }};
+      {% else %}
+        background: transparent;
+      {% endif %}
+
+      {% if page.show_accent_border %}
+        border-top: solid 8px #{{page.accent_border_color}};
+      {% endif %}
+    }
+
+    #primary-cta {
+      {% if page.button_background_color %}
+        background-color: #{{page.button_background_color}};
+        border-color: #{{page.button_background_color}};
+      {% endif %}
+
+      {% if page.button_text_color %}
+        color: #{{page.button_text_color}};
+      {% endif %}
+    }
+
+    .microsite-content a#primary-cta {
+      {% if page.button_text_color %}
+        color: #{{page.button_text_color}};
+      {% endif %}
+    }
+
+    /* TODO: Add hover color to model or dynamically generate */
+    #primary-cta:hover {
+      opacity: 0.9;
+      transition: 0.25s;
+    }
+
+    #primary_content {
+      background-size: cover;
+      {% if page.primary_content_background_color %}
+        background-color: #{{ page.primary_content_background_color }};
+      {% endif %}
+
+      {% if page.primary_content_text_color %}
+        color: #{{ page.primary_content_text_color }};
+      {% endif %}
+
+      {% if page.primary_content_background_image %}
+        {% image page.primary_content_background_image original as primary_img %}
+        background-image: url({{ primary_img.url }});
+      {% endif %}
+    }
+
+    #primary_content a {
+      {% if page.primary_content_text_color %}
+        color: #{{ page.primary_content_text_color }};
+      {% endif %}
+    }
+
+    #secondary_content {
+      background-size: cover;
+      {% if page.secondary_content_background_color %}
+        background-color: #{{ page.secondary_content_background_color }};
+      {% endif %}
+
+      {% if page.secondary_content_text_color %}
+        color: #{{ page.secondary_content_text_color }};
+      {% endif %}
+
+      {% if page.secondary_content_background_image %}
+        {% image page.secondary_content_background_image original as secondary_img %}
+        background-image: url({{ secondary_img.url }});
+      {% endif %}
+    }
+
+    #secondary_content a {
+      {% if page.secondary_content_text_color %}
+        color: #{{ page.secondary_content_text_color }};
+      {% endif %}
+    }
+
+    #footer, .microsite .footer {
```

Comment:
Why applying styles to regular footer?

Reply:
This should ideally be a separate field (main footer color and background) but did it quickly because we needed to keep regular footer but change the colors of it to match branding.

Reply:
I thought we either had standard footer, or custom footer. I did not think we were supporting standard footer but with different colors. If they want to change colors can use the custom footer instead?

Reply:
If we want to use regular OR footer with custom colors (which we do in this case) we can't do that without the raw HTML stream field addition because the footer layout is not re-createable 1-to-1 through the WYSIWYG editor. So we either need to add separate fields for OR footer values or add raw html streamfield support.

Reply:
I think it's fine if we want to support standard footer with custom colors, it's just not what I had in mind before so the model is confusing. We may want to clean up a bit at some point to polish this.

Reply:
👎  on raw html field 😛